### DISCUSSION TITLE
Domoticz: ESP12F_Relay_X8_v1.1 only 4 active channels

### DISCUSSION THREAD

**Discussion by Dylantje** [arendst/Tasmota#21066] [Q&A]
Did flash a:
[https://templates.blakadder.com/ESP12F_Relay_X8_v1_1.html](url)

Connect with wifi, mqtt with Domoticz
Only i can use unfortionly 4 channels i stead of 8
![ScreenShot070](https://github.com/arendst/Tasmota/assets/5214969/b87b1dc8-8afe-4b4b-882d-d4ed44efb8ab)

setting:
![image](https://github.com/arendst/Tasmota/assets/5214969/e732799a-e5a7-4802-8061-302acf6e99e4)

Where do i go wrong?

**Comment by sfromis** [comment]
The maximum for key/switch indexes in the config process is 4.

**Reply by Dylantje** [reply]
only the template says 8... How do i get then 8 channels working?

**Reply by sfromis** [reply]
The maximum is **really** 4 for the Domoticz integration.

Maybe you can do something handling the in/out topics yourself, but as I'm not using Domoticz, I have no specifics.

**Reply by barbudor** [reply]
@Dylantje What sfromis meant is that Domoticz driver do not support more than 4 key/switches

**Reply by barbudor** [reply]
As explained [here](https://github.com/arendst/Tasmota/issues/21019), when using USE_DOMOTICZ, you can't subscribe rules to the domoticz topics.
So you must disable USE_DOMOTICZ and do it all by yourself using Rules

**Reply by Dylantje** [reply]
ok so domoticz is the problem... CAn i use them with mqtt or someting?

**Reply by barbudor** [reply]
No Domonoticz is not the problem, Tasmota's driver for Domoticz is limited to 4

If you have node-red, then yes you can use NodeRED to "translate between Tasmota's topics and Domoticz Topics
In that case you don't need Domoticz in Tasmota at all

**Reply by sfromis** [reply]
The Tasmota integration with Domoticz is "old enough" for 4 relays to be generous, and the structure of the settings storage means that it is not simply to increase the max number.

**Reply by Dylantje** [reply]
Do you have a mqtt example for me? In node red
Switch on switch off, with a domoticz idx?

**Reply by arendst** [reply]
Let me think about a solution where Domoticz is still enabled in Tasmota but allows to use rules to expand input control ....

Let's continue here https://github.com/arendst/Tasmota/issues/21019
